### PR TITLE
Moved specs over from msgraph-metadata to be close to code

### PR DESCRIPTION
**Pull request by darrelmiller**
The permissions deployment model has been simplified to a flat list to now that it has been determined that we can't distringuish between versions here.  A permission cannot be deployed into one version versus the other and although we want to potentially disable a permission in v1 or beta for some URLs,. it doesn't mean we want to disable it for all URLs.  This means that we are going to have to support the option to include the version identifier in the path in our permissions description files.

### CODE REVIEW

File: specs/graph-permissions.cddl
Comment (wandeg) [inline specs/graph-permissions.cddl:1]: What does `cddl` stand for?

Comment (darrelmiller) [reply specs/graph-permissions.cddl:1]: It is a schema language for JSON and CBOR https://www.rfc-editor.org/rfc/rfc8610.html  It is similar to JSON Schema but just easier to read and write.  I just use it to help me design the format.

**Comment by athmangude** [comment]
@darrelmiller and @wandeg are we adding a schema to validate changes to the permissions-deployments file?

File: specs/permissions-deployment-schema.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{
+    "type": "object",
+    "properties": {
+        "permissionDeployments": {
+            "type": "object",
+            "patternProperties": {
+                "[\\w]+\\.[\\w]+[\\.[\\w]+]?": {
+                    "type": "object",
```
Comment (wandeg) [inline specs/permissions-deployment-schema.json:8]: Shouldn't this be an array of objects?

File: specs/permissions-deployment-schema.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+                    "properties": {
+                        "id": {
+                            "type": "string"
+                        },
+                        "resourceAppId": {
+                            "type": "string"
+                        },
+                        "environment": {
+                            "type": "string"
```
Comment (wandeg) [inline specs/permissions-deployment-schema.json:17]: Have we defined valid values for the `environment`?

File: specs/permissions-schema.json [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+            }
+        },
+        "pathSet": {
+            "type": "object",
+            "additionalProperties": false,
+            "properties": {
+                "schemeKeys": {
+                    "type": "array",
+                    "items": {
+                        "type": "string"
```
Comment (wandeg) [inline specs/permissions-schema.json:67]: Perhaps we could also list the valid values for the `schemeKeys` to aid in validation.

Comment (athmangude) [reply specs/permissions-schema.json:67]: I did that in this PR https://github.com/microsoftgraph/kibali/pull/41

File: specs/permissions-deployment-schema.json (hunk 1 above)
Comment (athmangude) [reply specs/permissions-deployment-schema.json:8]: Fixed in https://github.com/microsoftgraph/kibali/pull/42

File: specs/permissions-deployments.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+The "environment" member is a JSON string that identifies the deployment environment in which the permission SHOULD be supported.
+
+#### scheme member
+The "scheme" member is a JSON string that idenfies the type of authorization scheme that is enabled for the containing permission.
+
+#### isHidden
+The "isHidden" member is a boolean value that indicates if a permission should be publicly usable in the API.
+
+#### isDisabled
```
Comment (wandeg) [inline specs/permissions-deployments.md:68]: Can we change this to isEnabled? It's the same as what we have in the SP response https://learn.microsoft.com/en-us/graph/api/serviceprincipal-get?view=graph-rest-1.0&tabs=http#response-2

Comment (athmangude) [reply specs/permissions-deployments.md:68]: Let me update this in https://github.com/microsoftgraph/kibali/pull/42

Comment (darrelmiller) [reply specs/permissions-deployments.md:68]: Having isEnabled means that the property must be present in every deployment object.  If it was isDisabled then we only need to include where they are actually disabled, which should be just a small number.

File: specs/permissions-deployment-schema.json (hunk 2 above)
Comment (darrelmiller) [reply specs/permissions-deployment-schema.json:17]: We can open a separate PR for that once we know what the value values will be.  These values are fairly dynamic so I would suggest that we don't use an enum in the implementation.

**Review by wandeg — APPROVED** [review]

**Pull request merged**